src/convert.rs

```diff
@@ -152,10 +149,7 @@ pub trait ToPyArray {
     type Dim: Dimension;
 
     /// Deprecated form of [`ToPyArray::to_pyarray_bound`]
-    #[deprecated(
-        since = "0.21.0",
-        note = "will be replaced by `ToPyArray::to_pyarray_bound` in the future"
-    )]
+    #[cfg(feature = "gil-refs")]
```

Comment:
I think we probably want to keep these deprecation messages, and just add the new `#[cfg]` gates as well?

Reply:
Good idea, I will restore them once I get a chance.